File: 0867-transpose-matrix.java
/*
867. Transpose Matrix

Submitted: November 18, 2025

Runtime: 0 ms (beats 100.00%)
Memory: 46.44 MB (beats 95.07%)
*/

class Solution {
    public int[][] transpose(int[][] matrix) {
        int n = matrix.length;
        int m = matrix[0].length;
        int[][] res = new int[m][n];
        for (int i = 0; i < m; ++i) {
            for (int j = 0; j < n; ++j) {
                res[i][j] = matrix[j][i];
            }
        }
        return res;
    }
}

File: 0867-transpose-matrix.py
"""
867. Transpose Matrix

Submitted: November 18, 2025

Runtime: 3 ms (beats 19.35%)
Memory: 18.66 MB (beats 48.73%)
"""

class Solution:
    def transpose(self, matrix: List[List[int]]) -> List[List[int]]:
        return [
            [
                matrix[j][i]
                for j in range(len(matrix))
            ]
            for i in range(len(matrix[0]))
        ]